cpp/include/cudf_test/column_wrapper.hpp

```diff
@@ -1595,7 +1596,12 @@ class lists_column_wrapper : public detail::column_wrapper {
     thrust::copy_if(
       std::cbegin(cols), std::cend(cols), valids, std::back_inserter(children), thrust::identity{});
 
-    auto data = children.empty() ? cudf::empty_like(expected_hierarchy) : concatenate(children);
+    // TODO: Once the public concatenate API exposes streams, use that instead.
+    auto data =
+      children.empty()
+        ? cudf::empty_like(expected_hierarchy)
+        : cudf::detail::concatenate(
+            children, cudf::test::get_default_stream(), rmm::mr::get_current_device_resource());
```

Comment:
We should be aware that the changes in this PR that call `detail` APIs rather than public APIs will no longer create NVTX ranges. If we intend to keep those NVTX ranges, we should track the detail/public changes in an issue or TODO and change them back to use public methods once the public methods accept streams.

Reply:
Indeed, however I think this is the only such change. All other places where I added a `detail::` are just to disambiguate calls that were previously unambiguous but the compiler now cannot distinguish automatically because the public and detail copying APIs now share the same signature except for the default stream value, which is not sufficient to prefer one over the other (i.e. now `allocate_like` could be `cudf::detail::allocate_like` or `cudf::allocate_like` if used inside a `namespace cudf::detail {` block).

Reply:
Cool, I was hoping it was just disambiguation but wasn't 100% sure. I will resolve this.